lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp

```diff
@@ -1630,6 +1631,31 @@ chooseMfmaLikeStoreLayout(RankedTensorType valType) {
   return mfmaLL.compose(swapLL);
 }
 
+LinearLayout getPaddedRegToSharedLayout(const LinearLayout &regLayout,
+                                        PaddedSharedEncodingAttr paddedEnc) {
+  if (paddedEnc.getLinearComponent().has_value()) {
+    auto sharedLL = *paddedEnc.getLinearComponent();
+    // We cannot safely trim sharedLL so assert that regLayout's outdims are
+    // not smaller
+    for (auto [name, size] : regLayout.getOutDims()) {
+      assert(sharedLL.getOutDimSize(name) <= size);
+    }
+    SmallVector<int64_t> shape(regLayout.getOutDimSizes());
+    sharedLL = combineCtaCgaWithShape(*paddedEnc.getLinearComponent(),
+                                      paddedEnc.getCTALayout(), shape);
```

Comment:
I was just wondering whether the `LinearComponent` already has the `block` baked into it or not. Should we asset that it doesn't have it?